src/spikeinterface/extractors/tests/test_nwbextractors.py

```diff
@@ -287,5 +286,69 @@ def test_sorting_extraction_of_ragged_arrays(tmp_path):
     np.testing.assert_allclose(spike_train2, spike_times2)
 
 
+def test_sorting_extraction_start_time(tmp_path):
+    nwbfile = mock_NWBFile()
+
+    # Add the spikes
+    spike_times1 = np.array([0.0, 1.0, 2.0])
+    nwbfile.add_unit(spike_times=spike_times1)
+    spike_times2 = np.array([0.0, 1.0, 2.0, 3.0])
+    nwbfile.add_unit(spike_times=spike_times2)
+
+    file_path = tmp_path / "test.nwb"
+    # Write the nwbfile to a temporary file
+    with NWBHDF5IO(path=file_path, mode="w") as io:
+        io.write(nwbfile)
+
+    t_start = 10
+    sorting_extractor = NwbSortingExtractor(file_path=file_path, sampling_frequency=10.0, t_start=t_start)
+
+    extracted_spike_times1 = sorting_extractor.get_unit_spike_train(unit_id=0, return_times=True)
+    expected_spike_times1 = spike_times1 + t_start
+    np.testing.assert_allclose(extracted_spike_times1, expected_spike_times1)
+
+    extracted_spike_times2 = sorting_extractor.get_unit_spike_train(unit_id=1, return_times=True)
+    expected_spike_times2 = spike_times2 + t_start
+    np.testing.assert_allclose(extracted_spike_times2, expected_spike_times2)
+
+
+def test_sorting_extraction_start_time_from_series(tmp_path):
+    nwbfile = mock_NWBFile()
+    electrical_series_name = "ElectricalSeries"
+    t_start = 10.0
+
+    n_electrodes = 5
+    electrodes = mock_electrodes(n_electrodes=n_electrodes, nwbfile=nwbfile)
+    electrical_series = ElectricalSeries(
+        name=electrical_series_name,
+        starting_time=t_start,
+        rate=1.0,
+        data=np.ones((10, 5)),
+        electrodes=electrodes,
+    )
+    nwbfile.add_acquisition(electrical_series)
+    # Add the spikes
+    spike_times1 = np.array([0.0, 1.0, 2.0]) + t_start
+    nwbfile.add_unit(spike_times=spike_times1)
+    spike_times2 = np.array([0.0, 1.0, 2.0, 3.0]) + t_start
+    nwbfile.add_unit(spike_times=spike_times2)
+
+    file_path = tmp_path / "test.nwb"
+    # Write the nwbfile to a temporary file
+    with NWBHDF5IO(path=file_path, mode="w") as io:
+        io.write(nwbfile)
+
+    t_start = 10
+    sorting_extractor = NwbSortingExtractor(file_path=file_path, electrical_series_name=electrical_series_name)
+
+    extracted_spike_times1 = sorting_extractor.get_unit_spike_train(unit_id=0, return_times=True)
+    expected_spike_times1 = spike_times1 + t_start
+    np.testing.assert_allclose(extracted_spike_times1, expected_spike_times1)
```

Comment:
I think we should test frames rather than times. I'm ok if the returned times are not shifter by t_start (maybe it could be useful for syncrhonization), but frames should be offset with t_start.
```suggestion
    extracted_spike_times1 = sorting_extractor.get_unit_spike_train(unit_id=0)
    expected_spike_frames1 = np.round((spike_times1 - t_start) / sorting_extractor.sampling_frequency)
    np.testing.assert_allclose(extracted_spike_times1, expected_spike_times1)
```
We can maybe use a `rate=10`, so that the expected frames should be: [0, 10, 20]

Reply:
Fair enough, I added tests in this direction. I actuall don't know what the times should do.